scripts/fetchData.js

```diff
@@ -0,0 +1,27 @@
+import { getFromLocalStorage } from "./utils/localStorageUtil.js";
+
+export async function fetchData(url, options = { method: 'GET' }, shouldUseAuth = false,) {
+    try {
+        let fetchOptions = {
```

Comment:
The try/catch here will catch the error and just log it, so the function that uses this function won't be able to catch the error. I would advise removing the try/catch here, and using a try catch like you already do in the loginUser function.